hawkbit-repository/hawkbit-repository-jpa/src/main/java/org/eclipse/hawkbit/repository/jpa/JpaArtifactManagement.java

```diff
@@ -98,20 +99,33 @@ private static Artifact checkForExistingArtifact(final String filename, final bo
     public Artifact create(final ArtifactUpload artifactUpload) {
         final String filename = artifactUpload.getFilename();
         final long moduleId = artifactUpload.getModuleId();
-        AbstractDbArtifact result = null;
 
         final SoftwareModule softwareModule = getModuleAndThrowExceptionIfThatFails(moduleId);
 
-        final Artifact existing = checkForExistingArtifact(filename, artifactUpload.overrideExisting(),
-                softwareModule);
+        final Artifact existing = checkForExistingArtifact(filename, artifactUpload.overrideExisting(), softwareModule);
 
         assertArtifactQuota(moduleId, 1);
         assertMaxArtifactSizeQuota(filename, moduleId, artifactUpload.getFilesize());
         assertMaxArtifactStorageQuota(filename, artifactUpload.getFilesize());
 
+        AbstractDbArtifact result = getOrCreateArtifact(artifactUpload);
+        return result == null ? null : storeArtifactMetadata(softwareModule, filename, result, existing);
+    }
+
+    private AbstractDbArtifact getOrCreateArtifact(final ArtifactUpload artifactUpload) {
```

Comment:
Then you need to change this method correspondingly:
```suggestion
    private Optional<AbstractDbArtifact> getOrCreateArtifact(final ArtifactUpload artifactUpload) {
        final String providedSha1Sum = artifactUpload.getProvidedSha1Sum();

        if (!StringUtils.isEmpty(providedSha1Sum)) {
            final AbstractDbArtifact artifact = artifactRepository.getArtifactBySha1(tenantAware.getCurrentTenant(),
                    providedSha1Sum);
            if (artifact != null) {
                return Optional.of(artifact);
            }
        }

        return Optional.ofNullable(storeArtifact(artifactUpload));
    }
```